packages/babel-register/test/index.js

```diff
@@ -91,16 +93,60 @@ describe("@babel/register", function () {
     await new Promise(setImmediate);
 
     revertRegister();
-    currentHook = null;
-    currentOptions = null;
-    sourceMapSupport = false;
-    jest.resetModules();
   });
 
   afterAll(() => {
     resetCache();
   });
 
+  if (OLD_JEST_MOCKS) {
+    jest.doMock("pirates", () => mocks["pirates"]);
+    jest.doMock("source-map-support", () => mocks["source-map-support"]);
+
+    afterEach(() => {
+      jest.resetModules();
+    });
+  } else {
```

Comment:
This is the most annoying part of this PR: without Jest's complete `require` abstraction, we have to manually interact the `require` internals to mock modules.